voicemail/tests/voicemail.test.js

```diff
@@ -0,0 +1,111 @@
+let mockedTimeString = 'Thu, 13 Feb 2020 10:00:00 +0000';
+jest.mock('moment', () => {
+  const actualMoment = jest.requireActual('moment');
+  return () => {
+    return actualMoment(mockedTimeString);
+  };
+});
+
+const helpers = require('../../test/test-helper');
+const voicemailFunction = require('../functions/voicemail').handler;
+const Twilio = require('twilio');
+
+let context = {};
+let event = {};
+
+beforeAll(() => {
+  helpers.setup(context);
+});
+
+afterEach(() => {
+  context = {};
+  event = {};
+});
+
+afterAll(() => {
+  helpers.teardown();
+});
+
+test('returns a VoiceResponse', done => {
+  const callback = (err, result) => {
+    expect(result).toBeInstanceOf(Twilio.twiml.VoiceResponse);
+    done();
+  };
+
+  voicemailFunction(context, event, callback);
+});
+
+test('forwards calls during business hours', done => {
+  context = {
+    PATH: '/demo/voicemail',
+    MY_PHONE_NUMBER: '+1112223333',
+  };
+
+  const callback = (err, result) => {
+    expect(result.toString()).toMatch(
+      '<Response><Dial>+1112223333</Dial></Response>'
+    );
+    done();
+  };
+
+  voicemailFunction(context, event, callback);
+});
+
+test('handles timezone offset', done => {
+  context = {
+    PATH: '/demo/voicemail',
+    MY_PHONE_NUMBER: '+1112223333',
+    TIMEZONE_OFFSET: '-1',
+  };
+
+  mockedTimeString = 'Thu, 13 Feb 2020 18:59:00 +0000';
```

Comment:
This time string would fall within the work hours without the time offset. Would be good to see a test that had a UTC time out of work hours but the timezone offset puts it into work hours (or the opposite too).

Can never have too many timezone tests!